shell/platform/darwin/macos/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -0,0 +1,87 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "third_party/ocmock/Source/OCMock/OCMock.h"
+#include "flutter/shell/platform/darwin/macos/framework/Headers/FlutterEngine.h"
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#include "flutter/shell/platform/darwin/macos/framework/Source/FlutterDartProject_Internal.h"
+#include "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+#include "flutter/testing/testing.h"
+
+namespace flutter::testing {
+
+TEST(FlutterViewControllerTest, HasStrings) {
+  NSString* fixtures = @(testing::GetFixturesPath());
+  FlutterDartProject* project = [[FlutterDartProject alloc]
+      initWithAssetsPath:fixtures
+             ICUDataPath:[fixtures stringByAppendingString:@"/icudtl.dat"]];
+  FlutterViewController* viewController = [[FlutterViewController alloc] initWithProject:project];
+
+  // Mock getPasteboard so that this test will work in environments without a
+  // real pasteboard.
+  id pasteboardMock = OCMClassMock([NSPasteboard class]);
+  __block NSString* clipboardString = @"";
+  void (^setStringMock)(NSInvocation*) = ^(NSInvocation *invocation) {
+    [invocation getArgument:&clipboardString atIndex:2];
+  };
+  OCMStub([pasteboardMock setString:[OCMArg any] forType:[OCMArg any]]).andDo(setStringMock);
```

Comment:
This code would be a bit easier to follow if you inlined all these blocks instead of declaring them then passing them as variables then using the variables.